lib/hash_ring/ets.ex

```diff
@@ -252,21 +288,47 @@ defmodule ExHashRing.HashRing.ETS do
       {:reply, {:error, :node_exists}, state}
     else
       nodes = [{node_name, num_replicas} | nodes]
-      {:reply, {:ok, nodes}, rebuild(%{state | nodes: nodes})}
+      {:reply, {:ok, nodes}, update_nodes(state, nodes)}
+    end
+  end
+
+  def handle_call({:add_nodes, nodes}, _from, %__MODULE__{} = state) do
+    nodes = transform_nodes(nodes, state.default_num_replicas)
+
+    existing_nodes = Enum.filter(nodes, fn {name, _} ->
+      has_node_with_name?(state.nodes, name)
+    end)
+
+    if Enum.empty?(existing_nodes) do
+      nodes = nodes ++ state.nodes
+      {:reply, {:ok, nodes}, update_nodes(state, nodes)}
+    else
+      {:reply, {:error, :node_exists}, state}
     end
   end
 
   def handle_call({:remove_node, node_name}, _from, %{nodes: nodes} = state) do
     if has_node_with_name?(nodes, node_name) do
       nodes = Enum.reject(nodes, fn {existing_node, _} -> existing_node == node_name end)
-      {:reply, {:ok, nodes}, rebuild(%{state | nodes: nodes})}
+      {:reply, {:ok, nodes}, update_nodes(state, nodes)}
+    else
+      {:reply, {:error, :node_not_exists}, state}
+    end
+  end
+
+  def handle_call({:remove_nodes, node_names}, _from, %__MODULE__{} = state) do
+    unknown_nodes = Enum.reject(node_names, &has_node_with_name?(state.nodes, &1))
+
+    if Enum.empty?(unknown_nodes) do
+      nodes = Enum.reject(state.nodes, fn {name, _} -> name in node_names end)
+      {:reply, {:ok, nodes}, update_nodes(state, nodes)}
     else
       {:reply, {:error, :node_not_exists}, state}
```

Comment:
It seemed like a weird asymmetry to have `remove_node/2` return an error when given an unknown node but `remove_nodes/2` just ignores them.  

I find the behavior of add and remove to be a bit silly because they could easily be idempotent, which seems like a nice property, but I was taking the cues on how to behave here on how the current version works.

Reply:
I misread your comment the first time, returning the unknown nodes in addition to returning an error, yea it could do that.

I'm seriously considering changing this contract though to just be idempotent.

Reply:
I think it's in our best interest to error and not write so we don't accidentially F*** up the config by assuming state.

Reply:
Do you think it's best then to return the nodes that do not exist (in the remove case) and do exist (in the add case)?

Reply:
I think that makes the most sense. However if it's not trivial we don't have to do it... But writes should def error since they assume state